src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -140,115 +140,112 @@ public class HubEndPoint<THub, TClient> : EndPoint, IInvocationBinder where THub
 
         private async Task DispatchMessagesAsync(Connection connection)
         {
-            var invocationAdapter = _registry.GetInvocationAdapter(connection.Metadata.Get<string>("formatType"));
-
             // We use these for error handling. Since we dispatch multiple hub invocations
             // in parallel, we need a way to communicate failure back to the main processing loop. The
             // cancellation token is used to stop reading from the channel, the tcs
             // is used to get the exception so we can bubble it up the stack
             var cts = new CancellationTokenSource();
-            var tcs = new TaskCompletionSource<object>();
+            var completion = new TaskCompletionSource<object>();
 
             try
             {
                 while (await connection.Transport.Input.WaitToReadAsync(cts.Token))
                 {
                     while (connection.Transport.Input.TryRead(out var incomingMessage))
                     {
-                        InvocationDescriptor invocationDescriptor;
-                        var inputStream = new MemoryStream(incomingMessage.Payload);
-
-                        // TODO: Handle receiving InvocationResultDescriptor
-                        invocationDescriptor = await invocationAdapter.ReadMessageAsync(inputStream, this) as InvocationDescriptor;
-
-                        // Is there a better way of detecting that a connection was closed?
-                        if (invocationDescriptor == null)
+                        if (!_protocol.TryParseMessage(incomingMessage.Payload, this, out var hubMessage))
```

Comment:
As per the other comments - this will probably come later when we add support for protobuf so I think it is fine to go with JSON protocol hardcoded and tied to EndPoint and then fix it when adding protobuf